tests/Unit/Service/Persistance/PriorityServiceTest.php

```diff
@@ -26,39 +26,108 @@ public function setUp()
         $this->subject = new PriorityService($this->monitoringDataRepository->reveal());
     }
 
-    public function testCalculateNoPersistedDocument(): void
-    {
-        $monitoringDataDto = new MonitoringDataDto(
-            'id', 'satus', 'payload', 1, 60, new DateTimeImmutable(), 'root.branch.leaf', 5, '* 2'
-        );
-        $this->monitoringDataRepository->find('id')->shouldBeCalledOnce()->willReturn(null);
+    /**
+     * @dataProvider provideMonitoringData
+     */
+    public function testCalculate(
+        MonitoringDataDto $newMonitoringData,
+        ?MonitoringDataDocument $oldMonitoringData,
+        int $expectedPriority
+    ): void {
+        $this->monitoringDataRepository->find('id')->shouldBeCalledOnce()->willReturn($oldMonitoringData);
 
-        self::assertSame(1, $this->subject->calculate($monitoringDataDto));
+        self::assertSame($expectedPriority, $this->subject->calculate($newMonitoringData));
     }
 
-    public function testCalculateSameOrLowerPriority(): void
+    public function provideMonitoringData()
     {
-        $monitoringDataDto = new MonitoringDataDto(
-            'id', 'satus', 'payload', 15, 60, new DateTimeImmutable(), 'root.branch.leaf', 5, '* 2'
-        );
-        $monitoringData = new MonitoringDataDocument(
-            'id', 'satus', new DateTime('2019-01-01 00:00:00'), 'payload', 9, 60, new DateTimeImmutable(), 'root.branch.leaf', 5, '* 2'
-        );
-        $this->monitoringDataRepository->find('id')->shouldBeCalledOnce()->willReturn($monitoringData);
-
-        self::assertSame(15, $this->subject->calculate($monitoringDataDto));
+        return [
+            'old monitoring prio > new monitoring prio (both error)' => [
+                $newMonitoringData = $this->createMonitoringDataDto('error', 10),
+                $oldMonitoringData = $this->createMonitoringDataDocument('error', 20),
+                20
+            ],
+            'old monitoring prio = new monitoring prio (both error)' => [
+                $newMonitoringData = $this->createMonitoringDataDto('error', 20),
+                $oldMonitoringData = $this->createMonitoringDataDocument('error', 20),
+                20
+            ],
+            'old monitoring prio < new monitoring prio (both error)' => [
+                $newMonitoringData = $this->createMonitoringDataDto('error', 20),
+                $oldMonitoringData = $this->createMonitoringDataDocument('error', 10),
+                20
+            ],
+            'old monitoring prio OK > new monitoring prio ERROR' => [
```

Comment:
Hm, den Fall hätte ich anders erwartet. Der neue Push sollte die neue Prio setzen (genau wie im Beispiel direkt hiervor), oder nicht?

Reply:
Das ist der Theorie nach ein unmöglicher Fall. Es sei denn du hättest ursprünglich mit 20 gespusht und würdest jetzt über die API mit 10 pushen.

Wenn dich das stört kann ich
```
 && $oldMonitoringDataDocument->getStatus() === MonitoringStatus::ERROR()->getValue()
``` 
hinzufügen.

Dann verändert sich genau nur dieses Verhalten laut Tests.

Reply:
Hm, ok, eigentlich ist das ein Edge Case. Könnte nur passieren, wenn jemand den Pusher umkonfiguriert, während das Monitoring in `error` ist.
Würde ich so lassen, passt schon.